feat: add oauth

abhinavs1920 opened the pull request:
## PR Description

This pull request implements the core functionalities of the OAuth2 feature, specifically focusing on configuration saving and changes. The implementation utilizes the official Dart OAuth2 package and uses GitHub as the OAuth provider for illustration purposes. The Authorization Code flow type is currently implemented.

## Related Issues

- Closes #481

### Checklist
- [x] I have gone through the [contributing guide](https://github.com/foss42/apidash/blob/main/CONTRIBUTING.md)
- [x] I have updated my branch and synced it with the project main branch before making this PR
- [x] I am using the latest Flutter stable branch (run flutter upgrade and verify)
- [x] I have run the tests (flutter test) and all tests are passing

## Added/updated tests?
_We encourage you to add relevant test cases._

- [x] Yes
- [ ] No, and this is why: _please replace this line with details on why tests have not been included_

## OS on which you have developed and tested the feature?

- [x] Windows
- [ ] macOS
- [ ] Linux

ashitaprasad commented:
This PR contains a lot of unnecessary changes and shows lack of knowledge of existing code.
I recommend you to go through the entire API Dash codebase first to develop a better understanding.

abhinavs1920 commented:
@ashitaprasad Yep, actually that was my brute force approach, I just wanted to know if the flow I implemented currently is right or not. I will make a new PR with clean and optimized code with the current flow I implemented.
